tests/e2e/CMakeLists.txt

```diff
@@ -0,0 +1,226 @@
+
+############################################# Add E2E Tests ################################################
+
+
+set(FREERTOS_ROOT_PATH "${az_iot_middleware_freertos_SOURCE_DIR}/libraries/FreeRTOS/FreeRTOS")
+set(FREERTOS_PLUS_ROOT_PATH "${az_iot_middleware_freertos_SOURCE_DIR}/libraries/FreeRTOS/FreeRTOS-Plus")
+
+include_directories(${CMAKE_CURRENT_LIST_DIR}/../config_files)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m32")
+add_compile_options(-m32 -ggdb3 -O0 -D_DEBUG -DprojCOVERAGE_TEST=0 -DMBEDTLS_CONFIG_FILE=\"mbedtls_config.h\" -D_WINDOWS_)
```

Comment:
>D_WINDOWS_ [](http://example.com/codeflow?start=111&length=10)

Windows?

Reply:
these are all copied from the initial CMake that you had added. I wasn't sure if this was something you guys found was needed in Azure RTOS.

Reply:
now that we are using cmake ...lets remove these 
-m32 -ggdb3 -O0 -D_DEBUG

and do a top level build -DCMAKE_BUILD_TYPE=Debug and COMPILE_FLAGS "-m32" LINK_FLAGS "-m32"

Reply:
Yea the `-m32` compile flag is handled at the top. The debug option can be handled by the command line so removed that too. Check again to see if the remaining flags look alright. It built on my machine this way so we'll see how the CI likes it.